react/src/components/home/FilterButtons.jsx

```diff
@@ -1,197 +1,222 @@
+  });
+  const productOptions = [
+    "Fresh Produce",
+    "Dairy",
+    "Meat",
+    "Bread",
+    "Frozen Food",
+    "Baby Needs",
+    "Clothing",
+    "Medicine",
+  ];
+  const locationOptions = ["Food Pantry", "Soup Kitchen"];
+
+  useEffect(() => {
+    const dayParam = urlParams.get("d");
+    const productParam = urlParams.get("p");
+    const locationParam = urlParams.get("l");
+
+    if (dayParam) setFilteredDay({ selection: dayParam, selected: true });
+
+    if (productParam) setFilteredProduct({ selection: productParam, selected: true });
+
+    if (locationParam) setFilteredLocation({ selection: locationParam, selected: true });
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+
+  const handleClickOutside = (event) => {
+    const clickedOutsideLocationDropdown =
+      locationDropdownRef.current && !locationDropdownRef.current.contains(event.target);
+    const clickedOutsideDayDropdown =
+      dayDropdownRef.current && !dayDropdownRef.current.contains(event.target);
+    const clickedOutsideProductDropdown =
+      productDropdownRef.current && !productDropdownRef.current.contains(event.target);
+
+    if (
+      clickedOutsideLocationDropdown ||
+      clickedOutsideDayDropdown ||
+      clickedOutsideProductDropdown
+    ) {
+      setShowLocationDropdown(false);
+      setShowDayDropdown(false);
+      setShowProductDropdown(false);
+    }
+  };
+
+  const handleButtonClick = (dropdownSetter, otherDropdowns) => {
+    dropdownSetter((prev) => !prev);
+    otherDropdowns.forEach((setter) => setter(false));
+  };
+
+  const handleDayButton = () => {
+    if (!filteredDay.selected) {
+      handleButtonClick(setShowDayDropdown, [setShowLocationDropdown, setShowProductDropdown]);
+    } else {
+      handleRemoveFilter(setFilteredDay, "d");
+    }
+  };
+
+  const handleTypeButton = () => {
+    if (!filteredProduct.selected) {
+      handleButtonClick(setShowProductDropdown, [setShowLocationDropdown, setShowDayDropdown]);
+    } else {
+      handleRemoveFilter(setFilteredProduct, "p");
+    }
+  };
+
+  const handleLocationButton = () => {
+    if (!filteredLocation.selected) {
+      handleButtonClick(setShowLocationDropdown, [setShowDayDropdown, setShowProductDropdown]);
+    } else {
+      handleRemoveFilter(setFilteredLocation, "l");
+    }
+  };
+
+  const handleOptionClick = (option, type, setFiltered, setDropdown) => {
+    setFiltered({ selection: option, selected: true });
+    setDropdown(false);
+    urlParams.set(type, option);
+    navigate(`${location.pathname}?${urlParams.toString()}`);
+  };
+
+  const handleRemoveFilter = (setter, type) => {
+    setter({ selection: "", selected: false });
+    urlParams.delete(type);
+    navigate(`${location.pathname}?${urlParams.toString()}`);
+  };
+
+  return (
+    <>
+      <div className={`${!config.showDayOfTheWeekFilter && "d-none"}`}>
```

Comment:
Do you see how you had to negate "showDayOfTheWeekFilter"? This does not require changes, but for future thought, it might be more intuitive to name it "hideDayOfTheWeekFilter", so you don't have to use negated logic. One of those "more than one way to skin a cat" moments.

Reply:
Totally agree, I will definitely change it, it is more intuitive and clean.